src/renderer/components/dock/dock.store.ts

```diff
@@ -35,16 +35,24 @@ export enum TabKind {
   POD_LOGS = "pod-logs",
 }
 
-export interface IDockTab {
-  id?: TabId;
+export interface DockTabDescriptor {
+  id: TabId;
   kind: TabKind;
-  title?: string;
-  pinned?: boolean; // not closable
+  title: string;
+  pinned: boolean; // not closable
+
+  /**
+   * DockTabData may contain extra data
+   */
+  [key: string]: any;
 }
 
+export type CreateDockTabDescriptor = Partial<Omit<DockTabDescriptor, "kind">> & Pick<DockTabDescriptor, "kind">;
+export type SpecializedCreateDockTabDescriptor = Omit<CreateDockTabDescriptor, "kind">;
```

Comment:
Maybe specified type is better than `Partial<IDockTab>` but it's super verbose here. You can't say what it does and how it differs from `CreateDockTabDescriptor` when founding it somewhere in the code.

Reply:
It should throw an error if the `kind` field is specified. That is the desire because the functions that use this type specify the `kind` field.